### ISSUE TITLE
//regen generates completely different area

### ISSUE THREAD

**Issue by MsKiller01** [EngineHub/WorldEdit#1961]
### WorldEdit Version

7.2.8-snapshot-dist

### Platform Version

paper1.18-46

### Confirmations

- [X] I am using the most recent Minecraft release.
- [X] I am using a version of WorldEdit compatible with my Minecraft version.
- [X] I am using the latest or recommended version of my platform software.
- [X] I am NOT using a hybrid server, e.g. a server that combines Bukkit and Forge. Examples include Arclight, Mohist, and Cardboard.
- [X] I am NOT using a fork of WorldEdit, such as FastAsyncWorldEdit (FAWE) or AsyncWorldEdit (AWE)

### Bug Description

Normally, I think, //regen is using the world seed to regenerate a chosen chunk that comes real close to the original world and updates it to newly formed area.
This time when regenerate a chunk it comes up with a complete different area. Two diffenrent tests, first was one chunk on an ocean floor, where after the //regen thought there should be a mountain. Second test was in a puddle in an grassy field, yet after //regen a chunk it thought there was a deep oceanfloor.
Doing some tests below Y=0, there seems //chunk only takes the area from Y=-1 to Y=-64 and when above Y=0 it goes from Y=0 to Y=65000 or so, very high. Below Y=0 the //regen test showed not that much difference but taking this same chunk above Y=0 then it sees a complete different world and want to change it.

It is sad to see that you people have to bring out all your knowledge to tame the Mojang-beast that keeps coming at us (nice analogy I think) but I am convinced you can overcome these problems.

### Expected Behavior

The behaviour should fit very close to the original world, and a random chosen part of an ocean should not be regenerated as a big mountain almost straight up to the roof, it should blend in on the ocean floor.
The tests I did were on different parts of the world, old and newly generated chunks (not the complete world is yet explored)

### Reproduction Steps

1. ...go into the world as OPerator or equivalent and rights given to you for the use of Worldedit
2. ... do //chunk on any random part of the world
3. ... do //regen and see that the world becomes not what you expect to see (knowing that those parts were not what they showed after this action - ocean becomes sahara or flat grasslands becomes high steep mountains)
4. ... //undo (because we do not want this...)

### Anything Else?

no, I hope this is recognized by the team and if there is anything you want me to test, i'd be glad to be of assistance to you!

**Comment by wizjany** [comment]
regen is working fine here. I can only guess that you're trying to regen an old world or something.

**Comment by MsKiller01** [comment]
an old world....hmmm, no, that 'guess' is incorrect. But as I thought, this won't get fixed so it is time for me to tell people that their years of thousands of hours of builds has to go down the toilet because of a command that would not work the same anymore the way it did on 1.17.1. by beginning with a clean world.... 

Thanks for your tremendous effort of helping us all out!

**Comment by me4502** [comment]
The command does what it has always done, call the Minecraft world generator and replace the terrain with the output. If the terrain generator changes, what Minecraft gives WorldEdit will change. There have been no changes in the command, only changes in the Minecraft world generator.

**Comment by MsKiller01** [comment]
Please me4502, tell me where to look for this then, our world regen is not possible this way, since you are a programmer and I a humble server maintainer, I stumble on problems and try to look at them at various points.....

**Comment by me4502** [comment]
Given the intricacies of Minecraft world gen and how the code that generates it is copyrighted, we can't add support for old generations within WorldEdit.

The best you could probably do is regenerate on 1.17.1 and use a schematic to move it to the new world.

**Comment by MsKiller01** [comment]
Ok, thanks, are you also able to look into the //restore part of worldedit? it trips over and errors when the folder POI is in the backupfile, please?

**Comment by me4502** [comment]
That's already being worked on here, https://github.com/EngineHub/WorldEdit/pull/1959

**Comment by MsKiller01** [comment]
Thank you very much for your help, and this time I am really satisfied by your answers....  I mean it!

Happy holidays to you and your family (-:

**Comment by MsKiller01** [comment]
Levelseed seems to generate something completely different. I tested my level seed in a newly world generated with single player 1.17.1 and compared it to a single player 1.18 world.... toally different. So, no wonder that your //regen would generate a totally different piece..... It was wrong of me to think that you guys had something to do with this, because that was uncalled and unfoundend by me cause of lack of knowledge, but I had to start somewhere. So my appologies for being blunt...

Thanks anyway for your tremendoes work and help....
